### PR TITLE
Bug fix: do not interpret contigs with periods in their names as files in the interval parsing system

### LINKED ISSUE #1626: BaseRecalibrator interpreting contig name to a file
I was running BaseRecalibrator with an --interval of GL00207.1:1+ and received this error:

A USER ERROR has occurred: The file GL000207.1:1+ does not exist

It seems like if the contig name has a '.' in it, it thinks it should be looking for a file.  This worked for all other contigs that did not have a '.' in the name and it didn't complain about the Y or MT contigs that come before GL00207.1 in the full command:

org.broadinstitute.hellbender.tools.walkers.bqsr.BaseRecalibrator  --useOriginalQualities true --knownSites /cromwell_root/broad-gitc-reference/hg19/Homo_sapiens_assembly19.dbsnp138.vcf --knownSites /cromwell_root/broad-gitc-reference/hg19/Mills_and_1000G_gold_standard.indels.b37.sites.vcf --knownSites /cromwell_root/broad-gitc-reference/hg19/Homo_sapiens_assembly19.known_indels_20120518.vcf --output NA12878.recal_data.csv --intervals Y:1+ --intervals MT:1+ --intervals GL000207.1:1+ --intervals GL000226.1:1+ --intervals GL000229.1:1+ --intervals GL000231.1:1+ --intervals GL000210.1:1+ --intervals GL000239.1:1+ --intervals GL000235.1:1+ --intervals GL000201.1:1+ --intervals GL000247.1:1+ --intervals GL000245.1:1+ --intervals GL000197.1:1+ --intervals GL000203.1:1+ --intervals GL000246.1:1+ --intervals GL000249.1:1+ --intervals GL000196.1:1+ --intervals GL000248.1:1+ --intervals GL000244.1:1+ --intervals GL000238.1:1+ --intervals GL000202.1:1+ --intervals GL000234.1:1+ --intervals GL000232.1:1+ --intervals GL000206.1:1+ --intervals GL000240.1:1+ --intervals GL000236.1:1+ --intervals GL000241.1:1+ --intervals GL000243.1:1+ --intervals GL000242.1:1+ --intervals GL000230.1:1+ --intervals GL000237.1:1+ --intervals GL000233.1:1+ --intervals GL000204.1:1+ --intervals GL000198.1:1+ --intervals GL000208.1:1+ --intervals GL000191.1:1+ --intervals GL000227.1:1+ --intervals GL000228.1:1+ --intervals GL000214.1:1+ --intervals GL000221.1:1+ --intervals GL000209.1:1+ --intervals GL000218.1:1+ --intervals GL000220.1:1+ --intervals GL000213.1:1+ --intervals GL000211.1:1+ --intervals GL000199.1:1+ --intervals GL000217.1:1+ --intervals GL000216.1:1+ --intervals GL000215.1:1+ --intervals GL000205.1:1+ --intervals GL000219.1:1+ --intervals GL000224.1:1+ --intervals GL000223.1:1+ --intervals GL000195.1:1+ --intervals GL000212.1:1+ --intervals GL000222.1:1+ --intervals GL000200.1:1+ --intervals GL000193.1:1+ --intervals GL000194.1:1+ --intervals GL000225.1:1+ --intervals GL000192.1:1+ --intervals NC_007605:1+ --input broad-gotc-staging-storage/cromwell_execution/PairedEndSingleSampleWorkflow/124e075d-539f-47bd-b0ea-8e882a108cd6/call-AggregatedBamMarkDuplicates/NA12878.aligned.duplicates_marked.bam --reference broad-gitc-reference/hg19/Homo_sapiens_assembly19.fasta  --mismatches_context_size 2 --indels_context_size 3 --maximum_cycle_value 500 --mismatches_default_quality -1 --insertions_default_quality 45 --deletions_default_quality 45 --low_quality_tail 2 --quantizing_levels 16 --bqsrBAQGapOpenPenalty 40.0 --preserve_qscores_less_than 6 --defaultBaseQualities -1 --disable_all_read_filters false --interval_set_rule UNION --interval_padding 0 --readValidationStringency SILENT --secondsBetweenProgressUpdates 10.0 --disableSequenceDictionaryValidation false --createOutputBamIndex true --createOutputBamMD5 false --addOutputSAMProgramRecord false --help false --version false --verbosity INFO --QUIET false

This same command works with Gatk 3.5

### PR DESCRIPTION
**Pull request by droazen**
Resolves #1626

### CODE REVIEW

**Comment by droazen** [comment]
@lbergelson or @akiezun could you review?

File: src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java [hunk 1]
```diff
@@ -184,11 +184,6 @@ else if ( new File(arg).exists() ) {
                     "Please rename your file with the appropriate extension. If %s is NOT supposed to be a file, " +
                     "please move or rename the file at location %s", arg, arg, new File(arg).getAbsolutePath()));
         }
-        // If it's not an existing file, but looks like it was supposed to be a file (ie., it has an extension), throw an error.
-        // This works because '.' is not an allowable character in interval strings.
-        else if ( arg.indexOf('.') != -1 ) {
-            throw new UserException.CouldNotReadInputFile("The file " + arg + " does not exist");
-        }
         // Otherwise treat as an interval -> parse and add to raw interval list
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: so this was not true i guess "'.' is not an allowable character in interval strings." Can you add a comment about it to the code  ie state that '.' is in fact allowable?

**Comment by akiezun** [comment]
LGTM modulo comment in code. Merge when tests pass

File: src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java (hunk 1 above)
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/IntervalUtils.java]: Comment added

**Pull request merged**